app/components/moderator-list-row/component.js

```diff
@@ -0,0 +1,81 @@
+import { computed } from '@ember/object';
+import { inject as service } from '@ember/service';
+import Component from '@ember/component';
+
+import { task } from 'ember-concurrency';
+
+
+export default Component.extend({
+    i18n: service(),
+    store: service(),
+
+    removeConfirmation: false,
+    editConfirmation: false,
+    roleLabel: 'Role',
+    role: computed('moderator.permissionGroup', function() {
+        for (const value of this.get('roleOptions')) {
+            if (value.role === this.get('moderator.permissionGroup')) {
+                this.set('roleLabel', value.label);
```

Comment:
Yeah...the issue is I need role to be computed on the initial load and then set after that. I tried having it in `didRecieveAttrs` but that was triggered every time role changed. I think it would work in `init` though. Agree that it should be a computed.

Reply:
So could it be a [oneWay](https://www.emberjs.com/api/ember/3.0/functions/@ember%2Fobject%2Fcomputed/oneWay) (or its more understandable alias, `reads`)?
```js
role: reads('moderator.permissionGroup')
```